### PR TITLE
Prettying up yaml with setpoint limits, icons, etc...

### PR DESCRIPTION
**Pull request by barndawgie**
Moving https://github.com/Daniel-dev22/esphome-econet/pull/31 over to this repo.

### CODE REVIEW

File: econet_heatpump_water_heater.yaml [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+  friendly_name: "Water Heater"
   device_description: "Rheem Heat Pump Water Heater"
   tx_pin: GPIO19
   rx_pin: GPIO22
 
 esphome:
   name: ${name}
+  friendly_name: ${friendly_name}
   comment: ${device_description}
+  project:
```
Comment (tronikos) [inline econet_heatpump_water_heater.yaml:12]: Remove

File: econet_hvac.yaml [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+  friendly_name: "Rheem HVAC"
+  device_description: "Rheem HVAC"
   tx_pin: GPIO19
   rx_pin: GPIO22
 
 esphome:
   name: ${name}
+  friendly_name: ${friendly_name}
   comment: ${device_description}
+  project:
```
Comment (tronikos) [inline econet_hvac.yaml:12]: Remove

File: econet_tankless_water_heater.yaml [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+  friendly_name: "Tankless Water Heater"
   device_description: "Rheem Tankless Water Heater"
   tx_pin: GPIO19
   rx_pin: GPIO22
 
 esphome:
   name: ${name}
+  friendly_name: ${friendly_name}
   comment: ${device_description}
+  project:
```
Comment (tronikos) [inline econet_tankless_water_heater.yaml:12]: Remove

File: econet_tankless_water_heater.yaml [hunk 4]
```diff
@@ -47,11 +52,14 @@ econet:
...
     name: "Water Heater"
+    visual:
+      min_temperature: "43.3333"
+      max_temperature: "60"
+      temperature_step: 1.0f
 
 sensor:
   - platform: econet
     name: "Temperature In"
-    id: temp_in
```
Comment (tronikos) [inline econet_tankless_water_heater.yaml:54]: Revert

File: econet_tankless_water_heater.yaml [hunk 5]
```diff
@@ -60,7 +68,6 @@ sensor:
     entity_category: "diagnostic"
   - platform: econet
     name: "Temperature Out"
-    id: temp_out
```
Comment (tronikos) [inline econet_tankless_water_heater.yaml:63]: Revert

File: econet_tankless_water_heater.yaml [hunk 6]
```diff
@@ -90,7 +98,6 @@ sensor:
     entity_category: "diagnostic"
   - platform: econet
     name: "Flow Rate"
-    id: flow_rate
```
Comment (tronikos) [inline econet_tankless_water_heater.yaml:93]: Revert

File: econet_tankless_water_heater.yaml [hunk 7]
```diff
@@ -99,23 +106,21 @@ sensor:
...
       # Convert liters per minute to gpm
       - multiply: 0.264172
+    icon: "mdi:water"
   - platform: econet
     name: "Ignition Cycles"
     sensor_datapoint: IGNCYCLS
     unit_of_measurement: ""
     accuracy_decimals: 0
     entity_category: "diagnostic"
-  - platform: template
```
Comment (tronikos) [inline econet_tankless_water_heater.yaml:108]: Add this back

**Review by tronikos — CHANGES_REQUESTED** [review]

File: econet_heatpump_water_heater.yaml (hunk 1 above)
Comment (barndawgie) [reply econet_heatpump_water_heater.yaml:12]: Is `project` undesired for some reason? It still shows in the spec at https://esphome.io/components/esphome#esphome-creators-project

Comment (tronikos) [reply econet_heatpump_water_heater.yaml:12]: It isn't undesired. The project name you have is for the other repo. And this repo isn't currently using releases and it has no versions. Just drop it for now and we can add it back later with the correct name and version.

File: econet_heatpump_water_heater.yaml [hunk 8]
```diff
@@ -30,8 +35,8 @@ logger:
   tx_buffer_size: 2000
 
 external_components:
-  - source: github://stockmopar/esphome-econet
-  # - source: components
+    - source: github://stockmopar/esphome-econet
```
[the remark this answers is not in the document]
Comment (barndawgie) [reply econet_heatpump_water_heater.yaml]: I ran a code formatter which fixed this but altered/cleaned-up some other whitespace.

**Comment by barndawgie** [comment]
Thanks for the feedback and apologies for stomping on a few changes; I think I missed some drift between the dan22 and stokmopar repos. Should be cleaned up  now.

**Review by tronikos — APPROVED** [review]

**Pull request merged**